### DISCUSSION TITLE
Evaluation of prototype with users

### DISCUSSION THREAD

**Discussion by hojelse** [git-truck/git-truck#352] [General]
Template idea:

The user should be able to, by themselves:

- [x] Run the tool with `npx git-truck`
- [x] Install the tool globally with `npm i -g git-truck`
- [x] Use the tool from any of their git repositories
- [x] Specify a branch to analyse
- [ ] Specify a manual location to output the data to, relative to the current working directory
- [x] Specify any log level
- [x] Try out different chart types and metrics
- [x] Understand clearly what each chart type and metrics depicts

We could collect user evaluations in this discussion

Questions:
- What would you use this tool for?
- What did you learn from using the tool?

**Comment by emiljapelt** [comment]
Idea for questions

Research question (RQ)
1. Is the tool we made easy-to-use?
2. Did it increase understanding of the project evolution and the way developers contribute to the project?
3. Does it spark inspiration for how to prevent code base degradation?
4. For what persona/type of person is this most useful? (maybe extend this to question about: person type, team type, team organization, process type (e.g. scrum), git strategy, project type, tech stack)
5. In what context is it useful? (meetings / retrospective / review / figuring out who to ask about a portion of the codebase)
6. What is needed (features) before it could be useful enough to be a paid service? (maybe like free, team, enterprise -model)

Interview questions
- RQ1
   1. Did you have node and npm installed beforehand?
   1. Was anything confusing about the setup and run, process?
   1. Did you refer to the README at any time? If so, did you find what you where looking for?
   1. Did you at any point feel, something you wanted to do, was more difficult than need be?
- RQ2
   1. Did you learn something new about the evolution of a project you analyzed? If so, please point out with pictures and descriptions, what part of the program told you something new.
   1. Did you learn something new about the way developers contribute to the project?
- RQ3
   1. Did you discover a process-oriented quirk which could lead to code degradation?
   1. If so, how would you go about exploring it further or solving it?
- RQ4
   1. What is your job title?
   1. Describe your day-to-day job function and responsibilities.
- RQ5
   1. Which view did you find most useful?
   1. Why were they useful? Why were the others less useful?
   1. In what context would you revisit that view? (meetings, by yourself)
   1. What did you learn from using the tool, and with which metric? (Feel free to post screenshot and elaborate)
- RQ6
   1. Are there any metrics you would like to see added to the tool?

Technical questions, if stuff goes wrong, for debugging
- Did it work using npx?
- Did it work using a global install?
- What OS, in which version, were you running?
- What version of node were you using?
- What version of npm were you using?
- What version of git-truck did you use?
- What repo did you test, and on which branch?

**Comment by emiljapelt** [comment]
Interview Questions (IQs)

IF (THINGS_GO_WELL)
{

RQ1 - Is the tool we made easy-to-use?
 - Was anything about setting up git-truck confusing?
 - Was anything about exploring the git-truck UI confusing?
 - During the usage, did you ever refer to the README? 
	- If so, why, and did you find what you were looking for?

RQ2 - Did it increase understanding of the project evolution and the way developers contribute to the project?
 - Did you learn something interesting about a repository(s) you analyzed? 
	- If so, please point it out with descriptions and screenshots if possible.

RQ3 - Does it spark inspiration for how to prevent code base degradation?
 - Did you discover anything that could indicate code degradation in the repository(s)?
	- If so, please point it out with descriptions and screenshots if possible.

RQ4 - For what persona/type of person is this most useful? (maybe extend this to question about: person type, team type, team organization, process type (e.g. scrum), git strategy, project type, tech stack)
 - What is your relation to the repository(s)? (are you a developer, product owner, supervisor etc...?)
 - Describe your day-to-day function and responsibilities, relating to the repository(s).

RQ5 - In what context is it useful? (meetings / retrospective / review / figuring out who to ask about a portion of the code base)
 - Which views did you find the most useful and why, if any?
 - In what context would you revisit those views, if any? (meetings, individually)

RQ6 - What is needed (features) before it could be useful enough to be a paid service? (maybe like free, team, enterprise -model)
 - Are there any metrics/views that you would like to see added?

}
ELSE
{

 - Please describe what went wrong? Include pictures if possible.
 - Did you try using both npx and a global install?
 - What OS, in which version, were you running?
 - What version of node were you using?
 - What version of npm were you using?
 - What version of git-truck did you try?
 - What repository, on which branch, did you try?

}

**Reply by nimrossum** [reply]
@mircealungu What do you think about these questions?

**Reply by mircealungu** [reply]
It's easier to provide feedback in Docs: https://docs.google.com/document/d/1TEfDZpbWTBKIwz8Naqn1iIqQWuOtw7htKxCVOKixMkQ/edit?usp=sharing

**Reply by nimrossum** [reply]
@mircealungu 
I can't edit the document, so I have created a new one in our shared folder:
https://docs.google.com/document/d/1cjqIii9YRCmnI_YR64wLBGbfjcQ-VzMEsKn0IfJhYzs/edit
You should have access with your gmail account

**Comment by nimrossum** [comment]
Do you collaborate on git projects with other people?
Have you ever visualized your collaboration?
We have created git-truck 🚛 for this very purpose. 
If you want to try it out and help us in our research study, then please fill out this short questionnaire: [https://forms.gle/MqQeWokKiUrF3dRZ7](https://forms.gle/MqQeWokKiUrF3dRZ7?fbclid=IwAR26nEtfxid79qxOhT2ePppSmdiR43mNqU8lUCHaZ6i1Np6kHrmDPEIO43k)

**Reply by nimrossum** [reply]
Do you collaborate on git projects with other people?
Have you ever visualized your collaboration?
We have created git-truck 🚛 for this very purpose. 
Help us in our research study, by trying it and filling out a  short questionnaire: 
https://forms.gle/MqQeWokKiUrF3dRZ7
